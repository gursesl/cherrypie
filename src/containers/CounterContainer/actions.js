import * as c from './constants'

export function incrementAction() {
  return {
    type: c.INCREMENT_ACTION,
  }
}

export function decrementAction() {
  return {
    type: c.DECREMENT_ACTION,
  }
}

export function incrementActionAsync() {
  return {
    type: c.INCREMENT_ACTION_ASYNC,
  }
}
