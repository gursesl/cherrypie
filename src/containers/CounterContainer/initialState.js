import { fromJS } from 'immutable'

const initialState = fromJS({
  value: 0,
})

export default initialState
