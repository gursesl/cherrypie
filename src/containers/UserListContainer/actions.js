import * as c from './constants'

export function usersFetchStart() {
  return { type: c.USERS_FETCH_START }
}

export function usersFetchSuccess(users) {
  return {
    type: c.USERS_FETCH_SUCCESS,
    payload: users,
  }
}

export function usersFetchFailure(err) {
  return {
    type: c.USERS_FETCH_FAILURE,
    error: err,
  }
}
