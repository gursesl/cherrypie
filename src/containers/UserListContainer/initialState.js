import { fromJS } from 'immutable'
import { SELECTOR_USERS_USERS, SELECTOR_USERS_ERROR, SELECTOR_USERS_ISLOADING } from './constants'

const initialState = fromJS({
  [SELECTOR_USERS_USERS]: [
    {
      id: "95617189",
      firstName: "Elfrieda",
      lastName: "Frank",
      email: "Ada20@hotmail.com",
    },
    {
      id: "95617188",
      firstName: "Jim",
      lastName: "Smith",
      email: "jsmith@mail.com",
    },
  ],
  [SELECTOR_USERS_ERROR]: 'This is an error',
  [SELECTOR_USERS_ISLOADING]: true,
})

export default initialState
