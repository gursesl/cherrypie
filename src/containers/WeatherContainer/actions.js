import * as c from './constants'

export function weatherDataFetchStart() {
  return { type: c.WEATHER_DATA_FETCH_START }
}

export function weatherDataFetchSuccess(cities) {
  return {
    type: c.WEATHER_DATA_FETCH_SUCCESS,
    payload: cities,
  }
}

export function weatherDataFetchFailure(err) {
  return {
    type: c.WEATHER_DATA_FETCH_FAILURE,
    error: err,
  }
}
