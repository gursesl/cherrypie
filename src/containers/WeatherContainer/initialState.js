import { fromJS } from 'immutable'
import {
  SELECTOR_WEATHER_ZIP,
  SELECTOR_WEATHER_CITY,
  SELECTOR_WEATHER_RESULTS,
  SELECTOR_WEATHER_ERROR,
} from './constants'

const initialState = fromJS({
  [SELECTOR_WEATHER_ZIP]: '20165',
  [SELECTOR_WEATHER_CITY]: 'Sterling',
  [SELECTOR_WEATHER_RESULTS]: [],
  [SELECTOR_WEATHER_ERROR]: 'Load error',
})

export default initialState
