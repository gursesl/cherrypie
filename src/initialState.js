export default {
  users: [
    {
      id: "95617189",
      firstName: "Elfrieda",
      lastName: "Frank",
      email: "Ada20@hotmail.com"
    },
    {
      id: "95617188",
      firstName: "Jim",
      lastName: "Smith",
      email: "jsmith@mail.com"
    },
  ],
  error: null,
  isLoading: true,
  value: 10
}
