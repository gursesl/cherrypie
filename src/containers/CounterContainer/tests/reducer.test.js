// REFERENCE: REDUCER TEST
import immutable from 'immutable'
import initialState from '../../../initialState'
import counterContainerReducer from '../reducer'
import * as a from '../actions'

let state

describe('counterContainerReducer', () => {

  beforeEach(() => {
    state = immutable.fromJS(initialState)
  })

  it('should return the initial state', () => {
    expect(counterContainerReducer(undefined, {})).toEqual(state)
  })

  it('should handle incrementAction action correctly', () => {
    const expectedResult = state + 1
    expect(counterContainerReducer(state, a.incrementAction())).toEqual(expectedResult)
  })

  it('should handle decrementAction action correctly', () => {
    const expectedResult = state - 1
    expect(counterContainerReducer(state, a.decrementAction())).toEqual(expectedResult)
  })

  it('should handle incrementAsyncAction action correctly', () => {
    const expectedResult = state
    expect(counterContainerReducer(state, a.incrementActionAsync())).toEqual(expectedResult)
  })

  it('returns the initial state snapshot', () => {
    expect(counterContainerReducer(undefined, {})).toMatchSnapshot();
  });

  it('handles the incrementAction action snapshot', () => {
    expect(counterContainerReducer({}, a.incrementAction())).toMatchSnapshot();
  });

  it('handles the toggleNav action snapshot', () => {
    expect(counterContainerReducer({}, a.decrementAction())).toMatchSnapshot();
  });

  it('handles the toggleNav action snapshot', () => {
    expect(counterContainerReducer({}, a.incrementActionAsync())).toMatchSnapshot();
  });

})
