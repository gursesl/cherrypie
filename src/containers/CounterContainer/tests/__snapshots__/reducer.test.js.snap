// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counterContainerReducer handles the incrementAction action snapshot 1`] = `"[object Object]1"`;

exports[`counterContainerReducer handles the toggleNav action snapshot 1`] = `NaN`;

exports[`counterContainerReducer handles the toggleNav action snapshot 2`] = `Object {}`;

exports[`counterContainerReducer returns the initial state snapshot 1`] = `
Immutable.Map {
  "users": Immutable.List [],
  "error": null,
  "isLoading": true,
  "value": 10,
}
`;
